scripts/street_sign_recognizer.py

```diff
@@ -43,22 +104,26 @@ def sign_bounding_box(self):
         (left_top, right_bottom) where left_top and right_bottom are tuples of (x_pixel, y_pixel)
             defining topleft and bottomright corners of the bounding box
         """
-        # TODO: YOUR SOLUTION HERE
-        left_top = (200, 200)
-        right_bottom = (400, 400)
+        x, y, w, h = cv2.boundingRect(self.mask)
+        
+        left_top = (x, y)
+        right_bottom = (x + w, y + h)
         return left_top, right_bottom
 
     def run(self):
         """ The main run loop"""
         r = rospy.Rate(10)
         while not rospy.is_shutdown():
             if not self.cv_image is None:
-                print "here"
                 # creates a window and displays the image for X milliseconds
                 cv2.imshow('video_window', self.cv_image)
+                if (self.pred_total):
+                    if (self.pred_total[max(self.pred_total, key=self.pred_total.get)] > 5):
+                        self.pred_sign = max(self.pred_total, key=self.pred_total.get)
+                print self.pred_sign
```

Comment:
Instead of just printing the prediction every time, you could keep track of past predictions and build up a confidence that it is a certain sign, then print out the predicted sign once you have a high enough confidence that it is actually correct. This way you can accept a predictor which has a relatively high false-positive rate and still be able to get the sign right